runtime/lib/ttnn/operations/conv/conv2d.cpp

```diff
@@ -31,6 +43,21 @@ void run(const ::tt::target::ttnn::Conv2dOp *op, ProgramContext &context) {
           {op->padding_height(), op->padding_width()},
           {op->dilation_height(), op->dilation_width()}, op->groups(), bias,
           config));
+
+  // Workaround. The compiler models all ops as outputting to
+  // DRAM - interleaved. The compiler is not yet setup to model
+  // memory configs. TTNN::conv2d outputs to L1, in some sharded
+  // config. In order to ensure the compiler models the memory config
+  // "correctly", I put to_memory_config here. In addition to the
+  // compiler being "correct", subsequent eltwise ops require that both
+  // inputs be on DRAM - interleaved.
+  //
+  // Issue: https://github.com/tenstorrent/tt-mlir/issues/826
+  auto new_memconfig = out.memory_config();
+  new_memconfig.memory_layout = TensorMemoryLayout::INTERLEAVED;
+  new_memconfig.buffer_type = BufferType::DRAM;
+  out = ::ttnn::to_memory_config(out, new_memconfig, std::nullopt);
```

Comment:
Do we still need this hack to add `ttnn::to_memory_config` after [this is added in metal](https://github.com/tenstorrent/tt-metal/commit/f3c290acb0204e62889428c3c3c1de5082999ee8#diff-eb0cc75dd7731aa39f19e4cfb18d63f27e2fdd2a77a427a09cf7f52dc107b05bR868)